src/Controller/OG.php

```diff
@@ -27,25 +27,31 @@ class OG {
    *   The entity type.
    * @param $bundle
    *   The bundle name.
+   * @param $settings
+   *   (Optional) allow overriding the default definition of the field/instance.
+   *   Allowed values:
+   *   - field: override the field definitions.
```

Comment:
I think you are not following the D7 code here. You are not supposed to pass all the field values -- just the part you want to override. So you need in your code, to take the default values, and merge with the override settings.